app/assets/javascripts/initializers/initScrolling.js

```diff
@@ -212,6 +212,8 @@ function insertArticles(articles) {
   }
   articles.forEach(function insertAnArticle(article) {
     var existingEl = document.getElementById('article-link-' + article.id);
+    var currentUser = userData();
+    var currentUserId = currentUser && currentUser.id;
```

Comment:
**non-blocking suggestion:** Does [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) work here?

I can't recall if we couldn't use this in the asset pipeline vs when used inside `/app/javascript` (processed by webpacker), but it might help make this a bit nicer (personal take - not pushing hard for this) 🙂 

```
var currentUserId = currentUser?.id;
```

Reply:
I think I'm going to leave the code as-is; it's not verbose and i think the syntax is clearer. 🙏🏾

Reply:
Optional chaining won't work in these asset pipeline JS files 😢